ton_client/src/net/server_link.rs

```diff
@@ -407,31 +439,44 @@ impl ServerLink {
             headers.insert(name, value);
         }
 
-        let result = self
-            .client_env
-            .fetch(
-                &endpoint.query_url,
-                FetchMethod::Post,
-                Some(headers),
-                Some(request),
-                timeout,
-            )
-            .await;
+        let network_retries_count = self.config.network_retries_count;
+        let mut retry_count = 0;
+        'retries: loop {
+            let endpoint = if let Some(endpoint) = endpoint {
+                Arc::new(endpoint.clone())
+            } else {
+                self.state.get_query_endpoint().await?
+            };
 
-        if let Err(err) = &result {
-            if crate::client::Error::is_network_error(err) {
-                self.state.internal_suspend().await;
-                self.websocket_link.suspend().await;
-                self.websocket_link.resume().await;
+            let result = self
+                .client_env
+                .fetch(
+                    &endpoint.query_url,
+                    FetchMethod::Post,
+                    Some(headers.clone()),
+                    Some(request.clone()),
+                    query.timeout,
+                )
+                .await;
+
+            if let Err(err) = &result {
+                if crate::client::Error::is_network_error(err) {
+                    self.state.internal_suspend().await;
+                    self.websocket_link.suspend().await;
+                    self.websocket_link.resume().await;
+                    retry_count += 1;
+                    if retry_count <= network_retries_count {
+                        continue 'retries;
```

Comment:
Why don't use just `continue`?

Reply:
Labels used with break and continue makes code clearer and more readable especially in large loops.
In this case `continue retries` is clearer than just `continue`